site/app/views/forum/ForumThreadView.php

```diff
@@ -997,97 +947,56 @@ public function createPost($thread_id, $post, $unviewed_posts, $function_date, $
-			$this->core->redirect($this->core->buildUrl(array('component' => 'navigation')));
-			return;
-		}
-
-		$this->core->getOutput()->addBreadcrumb("Discussion Forum", $this->core->buildUrl(array('component' => 'forum', 'page' => 'view_thread')));
-		$this->core->getOutput()->addBreadcrumb("Create Thread", $this->core->buildUrl(array('component' => 'forum', 'page' => 'create_thread')));
-		$return = <<<HTML
-		<script type="text/javascript" language="javascript" src="{$this->core->getConfig()->getBaseUrl()}vendor/jquery.are-you-sure/jquery.are-you-sure.js"></script>
-		<script type="text/javascript" src="{$this->core->getConfig()->getBaseUrl()}js/drag-and-drop.js"></script>
-		<script> 
-			$( document ).ready(function() {
-				enableTabsInTextArea("[name=thread_post_content]");
-				$("#thread_form").submit(createThread);
-				$("form").areYouSure();
-			});
-		 </script>
-HTML;
-        if($this->core->getUser()->getGroup() <= 3){
-            $categories = $this->core->getQueries()->getCategories();
-
-            $dummy_category = array('color' => '#000000', 'category_desc' => 'dummy', 'category_id' => "dummy");
-            array_unshift($categories, $dummy_category);
-
-            $return .= $this->core->getOutput()->renderTwigTemplate("forum/CategoriesForm.twig", [
-                "categories" => $categories,
-                "category_colors" => $category_colors
-            ]);
+        if(!$this->forumAccess()){
+            $this->core->redirect($this->core->buildUrl(array('component' => 'navigation')));
+            return;
         }
 
-		$return .= <<<HTML
-		<div class="content forum_content">
-		
-HTML;
-
-	$buttons = array(
-		array(
-			"required_rank" => 4,
-			"display_text" => 'Back to Threads',
-			"style" => 'position:relative;top:3px;float:right;',
-			"link" => array(true, $this->core->buildUrl(array('component' => 'forum', 'page' => 'view_thread'))),
-			"optional_class" => '',
-			"title" => 'Back to threads',
-			"onclick" => array(false)
-		)
-	);
-
-	$thread_exists = $this->core->getQueries()->threadExists();
+        $this->core->getOutput()->addBreadcrumb("Discussion Forum", $this->core->buildUrl(array('component' => 'forum', 'page' => 'view_thread')));
+        $this->core->getOutput()->addBreadcrumb("Create Thread", $this->core->buildUrl(array('component' => 'forum', 'page' => 'create_thread')));
 
-	$return .= $this->core->getOutput()->renderTwigTemplate("forum/ForumBar.twig", [
-								"forum_bar_buttons_right" => $buttons,
-								"forum_bar_buttons_left" => [],
-								"show_threads" => false,
-								"thread_exists" => $thread_exists,
-								"show_more" => false
-	]);
+        $this->core->getOutput()->addInternalJs('drag-and-drop.js');
+        $this->core->getOutput()->addVendorJs('flatpickr/flatpickr.js');
+        $this->core->getOutput()->addVendorJs('jquery.are-you-sure/jquery.are-you-sure.js');
 
+        $this->core->getOutput()->addInternalCss('flatpickr.min.css');
```

Comment:
this should be `$this->core->getOutput()->addVendorCss('flatpickr/flatpickr.min.css');`